### PR TITLE
Batching triples.py and symplectics.py 

### PR DESCRIPTION
**Pull request by apchytr**
**Context:** Enabling batching the methods that generate our built-in CircuitComponents i.e. `triples.py` and `symplectics.py`. 

Merge [Backend stack and broadcast to](https://github.com/XanaduAI/MrMustard/pull/561) before this PR ✅ 

**Description of the Change:** Introduced the `triples_batched.py` file. Updated `symplectics.py`. Misc changes.

**Benefits:** Arbitrary batch dim triples and symplectics

### CODE REVIEW

File: mrmustard/physics/utils.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+from mrmustard import math
+
+
+#  ~~~~~~~~~
+#  Utilities
+#  ~~~~~~~~~
+
+
+def compute_batch_size(*args) -> tuple[tuple[int, ...] | None, int]:
```
Comment (apchytr) [inline mrmustard/physics/utils.py:28]: Can probably replace with `broadcast_arrays`

File: mrmustard/lab_dev/transformations/sgate.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def __init__(
             modes_in=(mode,),
             modes_out=(mode,),
             ansatz=PolyExpAnsatz.from_function(
-                fn=triples.squeezing_gate_Abc, r=self.parameters.r, delta=self.parameters.phi
+                fn=triples.squeezing_gate_Abc,
+                r=self.parameters.r,
+                delta=self.parameters.phi,
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/sgate.py:97]: maybe we should use the same name

File: mrmustard/math/backend_jax.py [hunk 3]
```diff
@@ -77,22 +77,20 @@ def abs(self, array: jnp.ndarray) -> jnp.ndarray:
     def any(self, array: jnp.ndarray) -> jnp.ndarray:
         return jnp.any(array)
 
+    @partial(jax.jit, static_argnames=["start", "limit", "delta", "dtype"])
```
Comment (hitarth64) [inline mrmustard/math/backend_jax.py:80]: since all the input arguments are static, it may not be worth doing a jit of this specific function.

Comment (hitarth64) [reply mrmustard/math/backend_jax.py:80]: Not applicable. Please keep the jit decorator.

File: tests/test_physics/test_triples.py [hunk 4]
```diff
@@ -22,26 +22,39 @@
...
-        with pytest.raises(ValueError, match=match):
-            triples.displaced_squeezed_vacuum_state_Abc([1, 2], [3, 4, 5], 6, 7)
+    #     with pytest.raises(error, match=match):
+    #         triples.coherent_state_Abc([1, 2], [3, 4, 5])
+
+    #     with pytest.raises(error, match=match):
+    #         triples.squeezed_vacuum_state_Abc([1, 2], [3, 4, 5])
+
+    #     with pytest.raises(error, match=match):
+    #         triples.displaced_squeezed_vacuum_state_Abc([1, 2], [3, 4, 5], 6, 7)
```
Comment (ziofil) [inline tests/test_physics/test_triples.py:57]: remove commented code

File: tests/test_physics/test_triples.py [hunk 5]
```diff
@@ -375,19 +405,22 @@ def test_XY_to_channel_Abc(self):
...
         eta = np.random.random()
-        X = np.sqrt(eta) * np.eye(2)
-        Y = settings.HBAR / 2 * (1 - eta) * np.eye(2)
+        X = math.sqrt(eta) * math.eye(2)
+        Y = settings.HBAR / 2 * (1 - eta) * math.eye(2)
 
         A, b, c = triples.XY_to_channel_Abc(X, Y)
 
-        A_by_hand = np.block(
+        A_by_hand = math.concat(
```
Comment (ziofil) [inline tests/test_physics/test_triples.py:413]: we have math.block in case you prefer

File: mrmustard/physics/symplectics.py [hunk 6]
```diff
@@ -47,7 +62,22 @@ def czgate_symplectic(s: float) -> Matrix:
...
     """
-    return math.astensor([[1, 0, 0, 0], [0, 1, 0, 0], [0, s, 1, 0], [s, 0, 0, 1]])
+    s = math.astensor(s, dtype=math.complex128)
+    batch_shape = s.shape
+    batch_dim = len(batch_shape)
+
+    O_matrix = math.zeros(batch_shape, math.complex128)
+    I_matrix = math.ones(batch_shape, math.complex128)
+
+    return math.stack(
```
Comment (ziofil) [inline mrmustard/physics/symplectics.py:72]: why not math.block?

**Review by ziofil — DISMISSED** [review]
awesome. Pre-approved.

**Review by ziofil — APPROVED** [review]
great!
